src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -860,34 +875,53 @@ private static GapicProductConfig createDummyInstance(
     return ImmutableMap.copyOf(mergedResourceNameConfigs);
   }
 
-  private static ImmutableMap<String, SingleResourceNameConfig> createSingleResourceNameConfigs(
+  private static void createSingleAndFixedResourceNameConfigs(
       DiagCollector diagCollector,
       ConfigProto configProto,
+      ImmutableMap.Builder<String, SingleResourceNameConfig> singleResourceNamesBuilder,
+      ImmutableMap.Builder<String, FixedResourceNameConfig> fixedResourceNamesBuilder,
       @Nullable List<ProtoFile> sourceProtos,
       TargetLanguage language) {
     ProtoFile file = null;
     if (sourceProtos != null) {
       file = sourceProtos.get(0);
     }
+    List<CollectionConfigProto> allCollectionConfigProtos =
+        new ArrayList<>(configProto.getCollectionsList());
+    configProto
+        .getInterfacesList()
+        .stream()
+        .forEach(i -> allCollectionConfigProtos.addAll(i.getCollectionsList()));
+
     LinkedHashMap<String, SingleResourceNameConfig> singleResourceNameConfigsMap =
         new LinkedHashMap<>();
-    for (CollectionConfigProto collectionConfigProto : configProto.getCollectionsList()) {
-      createSingleResourceNameConfig(
-          diagCollector, collectionConfigProto, singleResourceNameConfigsMap, file, language);
-    }
-    for (InterfaceConfigProto interfaceConfigProto : configProto.getInterfacesList()) {
-      for (CollectionConfigProto collectionConfigProto :
-          interfaceConfigProto.getCollectionsList()) {
+    for (CollectionConfigProto collectionConfigProto : allCollectionConfigProtos) {
+      PathTemplate pathTemplate = PathTemplate.create(collectionConfigProto.getNamePattern());
+      try {
+        pathTemplate.instantiate(new HashMap<>());
+      } catch (ValidationException e) {
+        // This must be a single resource name, because it has binding vars in the path pattern.
         createSingleResourceNameConfig(
             diagCollector, collectionConfigProto, singleResourceNameConfigsMap, file, language);
+        continue;
       }
-    }
 
-    if (diagCollector.getErrorCount() > 0) {
```

Comment:
Looks like we are dropping this check. Should we keep it, since the `createSingleResourceNameConfig` function accepts the diagCollector?

Reply:
I'll bring it back, but I would rather it throw an error at that point. (If there are errors at this point, there's been enough time to collect a fair point, and generation won't succeed anyway. It's kinda annoying when the generator just spits out literally nothing instead of giving a useful error message.)